moped-editor/src/views/projects/projectView/ProjectNameComplex.js

```diff
@@ -0,0 +1,307 @@
+
+  // used to control the snackbar's contents and visibility
+  const [snackbarState, setSnackbarState] = useState(DEFAULT_SNACKBAR_STATE);
+
+  // apollo hook returning the async function to update the project names
+  const [updateProjectNames] = useMutation(UPDATE_PROJECT_NAMES_QUERY);
+
+  // this function is fired when the user clicks on the check mark to save the names
+  const handleAcceptClick = (e) => {
+    e.preventDefault();
+
+    // Check if the primary name is defined and has meaningful content
+    if (!projectName || projectName.trim() === "") {
+      // if it does not, set the error state
+      setPrimaryTitleError(true);
+    } else {
+      // and if it passes the test, clear any error state and save it via a mutation
+      setPrimaryTitleError(false);
+      updateProjectNames({
+        variables: {
+          projectId: props.projectId,
+          projectName: projectName,
+          projectNameSecondary: secondaryName,
+        },
+      })
+        .then((res) => {
+          // if the mutation is successful, show a success snackbar
+          setSnackbarState({
+            open: true,
+            message: <span>Success! The project names have been updated!</span>,
+            severity: "success",
+          });
+        })
+        .catch((error) => {
+          // and if it fails, show an error snackbar
+          setSnackbarState({
+            open: true,
+            message: (
+              <span>Error. The project names have not been updated.</span>
+            ),
+            severity: "error",
+          });
+        })
+        .finally(() => {
+          // return to the view mode, clear the snackbar after 3 seconds,
+          // and alert the parent component of the change
+          props.setIsEditing(false);
+          setTimeout(() => setSnackbarState(DEFAULT_SNACKBAR_STATE), 3000);
+          props.updatedCallback();
+        });
+    }
+  };
+
+  // this is fired when the user clicks on the 'X' to cancel the edit
+  const handleCancelClick = (e) => {
+    e.preventDefault();
+    props.setIsEditing(false);
+  };
+
+  // this is fired onChange of the project name field to check if it's valid
+  const handleProjectNameChange = (e) => {
+    // Update the state variables when the input changes
+    setProjectName(e.target.value);
+    if (!e.target.value) {
+      setPrimaryTitleError(true);
+    } else {
+      setPrimaryTitleError(false);
+    }
+  };
+
+  const handleSecondaryNameChange = (e) => {
+    // Update the secondary name state variable when the input changes
+    setSecondaryName(e.target.value);
+  };
+
+  // this is fired when the snackbar's close button is clicked or by timeout
+  const handleSnackbarClose = (event, reason) => {
+    setSnackbarState(DEFAULT_SNACKBAR_STATE);
+  };
+
+  // Because the layout is non-trivially different between view and edit modes,
+  // we're going to use two different JSX blocks to render the component's content.
+  const viewModeJSX = (
+    <>
```

Comment:
I dont think you need this react fragment tag

Reply:
Sounds great, I've removed both of these empty tag pairs. Thanks! https://github.com/cityofaustin/atd-moped/pull/1302/commits/fee13475f278b7a5f914281d9e99a0ef0bd21165